### PR TITLE
Migrate Java Binding to Maven Central Repository

### PR DESCRIPTION
**Pull request by laves**
- changed binding project to Gradle build project
- removed IntelliJ dependencies
- limited built-in keywords to standard list
- added publish script to publish to OSSRH staging

### CODE REVIEW

File: binding/java/build.gradle [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    mavenCentral()
+}
+
+sourceSets {
+    main {
+        java {
+            srcDirs = ['src']
+        }
+    }
+    test{
```
Comment (dbartle) [inline binding/java/build.gradle:24]: style inconsistency. format this with VS Code?

File: binding/java/publish-mavencentral.gradle [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    license {
+                        name = 'Apache License - v2.0'
+                        url = 'https://github.com/Picovoice/porcupine/blob/master/LICENSE'
+                    }
+                }
+                developers {
+                    developer {
+                        id = 'ilavery'
+                        name = 'Ian Lavery'
+                        email = 'ian@picovoice.ai'
```
Comment (dbartle) [inline binding/java/publish-mavencentral.gradle:51]: Should be generic info (Picovoice / hello@picovoice.ai), or omitted from the repo

File: demo/java/README.md [hunk 3]
```diff
@@ -24,17 +24,18 @@ Porcupine is:
...
 ## Installation
 
-You can get the latest Java demo executable JARs [here](/demo/java/bin). 
-
-If you wish, you can build the demos from source by opening the project with the [IntelliJ IDE](https://www.jetbrains.com/idea/download/).
-Select "Build > Build Project" to build the two demo classes or "Build > Build Artifacts" to create the executable JARs.
+Build the demo jars with Gradle:
+```console
+cd porcupine/demo/java
+./gradlew build
```
Comment (dbartle) [inline demo/java/README.md:30]: chmod +x gradlew

File: demo/java/build.gradle [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+}
+
+dependencies {
+    implementation 'ai.picovoice:porcupine-java:1.9.0'
+    implementation 'commons-cli:commons-cli:1.4'
+}
+
+jar {
```
Comment (dbartle) [inline demo/java/build.gradle:25]: this should mirror the file demo, but the setup differs

File: binding/java/publish-mavencentral.gradle [hunk 5]
```diff
@@ -0,0 +1,80 @@
+apply plugin: 'java-library'
```
Comment (dbartle) [inline binding/java/publish-mavencentral.gradle:1]: I don't believe any of this file should be checked in?

**Review by dbartle — CHANGES_REQUESTED** [review]

File: demo/java/build.gradle (hunk 4 above)
Comment (laves) [reply demo/java/build.gradle:25]: only way to disable the main JAR build that I could find was to overwrite it with another operation. This prevents a useless jar from cluttering up the build directory.

**Pull request merged**